proposals/lip-0070.md

```diff
@@ -384,7 +384,7 @@ def updateSharedRewards(generatorAddress: Address, tokenID: TokenID, reward: uin
     sharingCoefficientIncrease = muldiv_96(reward, rewardFraction, totalStake)
     # Lock the amount that needs to be shared.
     sharedRewards = mul_96(sharingCoefficientIncrease, sub_96(totalStake, selfStake))
-    sharedRewards = Q_96_ToInt(sharedRewards)
+    sharedRewards = min(Q_96_ToIntRoundUp(sharedRewards), reward)      # shared rewards can not exceed total reward in case of rounding issue
```

Comment:
Note that `reward` was overwritten above to be a Q96 number (note that overwriting input parameters to a function is generally considered bad practice). But for the minimum here we actually would need the original reward.

Reply:
Further, is the minimum really necessary. Don't we have
```
sharedRewards = (reward * rewardFraction / totalStake ) * (totalStake - selfStake)
```
where `rewardFraction <=1` and `(totalStake - selfStake) / totalStake <= 1` and we always round down intermediate results.

Reply:
In the implementation we have 
https://github.com/LiskHQ/lisk-sdk/blob/development/framework/src/modules/pos/method.ts#L121-L124

but i also cannot think of a case where sharedReward exceeding the original reward

Reply:
Removed the over-write of `reward` variable (it was done mostly for readability reasons but was not so necessary anyways, since it is not used that much..).

As for the min operator, I think in some test scenarios with 0 commission and only one voter we had some rounding issue (@shuse2 not sure I am right..) so we decided to add it for extra security. Fine for me to skip it, but would prefer to have it here to avoid any extra bugs.

Reply:
Im fine to keep the `min` since it will not harm a

Reply:
Sounds good, can be resolved then.